python/py_unzip/rezipper.py

```diff
@@ -54,7 +54,11 @@ def run(src: str, dst: str, package_dir: pathlib.Path, main: pathlib.Path):
                 if info.filename == "__main__.py":
                     addfile(info.filename, main.read_bytes(), mode=0o755)
                 else:
-                    addfile(info.filename, z_in.read(info.filename), mode=0o644)
+                    addfile(
+                        info.filename,
+                        z_in.read(info.filename),
+                        mode=info.external_attr >> 16 or 0o644,
```

Comment:
Not sure there's a use case but this would disallow `0o000` permissions right?

Reply:
hmm... do we have any use-cases for shipping inaccessible files as part of one of these applications? We use `0o000` for testing purposes, but I don't know of any places where we rely on that at product runtime

Reply:
Maybe not, it's a bit of an oddball set of permissions, but might be worth a cursory check on a DUT with something like `rpm -qla | xargs stat -c '%a %n | grep 000`